src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -501,6 +506,11 @@ public bool ContainsSyntaxTree(SyntaxTree syntaxTree)
 
         protected abstract bool CommonContainsSyntaxTree(SyntaxTree? syntaxTree);
 
+        /// <summary>
+        /// Optional semantic model provider for this compilation.
+        /// </summary>
+        internal readonly SemanticModelProvider? SemanticModelProvider;
```

Comment:
>internal [](start = 8, length = 8)

We don't want to expose it to the public?

Reply:
I see, it is itself internal

---
In reply to: [465317125](https://github.com/dotnet/roslyn/pull/46508#discussion_r465317125) [](ancestors = 465317125)